packages/spector/src/cli/cli.ts

```diff
@@ -14,9 +15,56 @@ import { getCommit } from "../utils/misc-utils.js";
 
 export const DEFAULT_PORT = 3000;
 
+function copyDir(src: string, dest: string) {
+  if (!fs.existsSync(src)) {
+    throw new Error(`Source directory not found: ${src}`);
+  }
+
+  if (!fs.existsSync(dest)) {
+    fs.mkdirSync(dest, { recursive: true });
+  }
+
+  const entries = fs.readdirSync(src, { withFileTypes: true });
+
+  for (const entry of entries) {
+    const srcPath = path.join(src, entry.name);
+    const destPath = path.join(dest, entry.name);
+
+    // If the entry is a directory, copy it recursively
+    if (entry.isDirectory()) {
+      copyDir(srcPath, destPath);
+    } else {
+      // If it's a file, copy the file
+      fs.copyFileSync(srcPath, destPath);
+    }
+  }
+}
+
+function copyToTemp(scenariosPaths: string[], tempFolder: string, destFolder: string) {
+  const tempDir = resolve(process.cwd(), `./temp/${tempFolder}/${destFolder}`);
+  if (!fs.existsSync(tempDir)) {
+    fs.mkdirSync(tempDir, { recursive: true });
+  }
+
+  for (const scenarioPath of scenariosPaths) {
+    const dir = resolve(process.cwd(), `${path.dirname(scenarioPath)}/${destFolder}`);
+    if (!fs.existsSync(dir)) continue;
+    copyDir(dir, tempDir);
+  }
+  return tempDir;
+}
+
+function copyScenariosToTemp(scenariosPaths: string[]) {
+  const tempDirname = `t${Math.floor(Math.random() * 10000)}`;
+  copyToTemp(scenariosPaths, `${tempDirname}`, `dist`);
+  copyToTemp(scenariosPaths, `${tempDirname}`, `assets`);
+  const tempDir = copyToTemp(scenariosPaths, `${tempDirname}`, `specs`);
+  return tempDir;
+}
+
 async function main() {
   await yargs(process.argv.slice(2))
-    .scriptName("spec")
+    .scriptName("tsp-spector")
```

Comment:
This is not really related to this change, But, I believe we could do this as part of this PR. Thanks